tests/e2e/utils/DriverHelper.ts

```diff
@@ -88,41 +88,49 @@ export class DriverHelper {
     }
 
     public async waitVisibility(elementLocator: By, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT): Promise<WebElement> {
-        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
         const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+        const attempts: number = Math.ceil(timeout / polling);
 
         Logger.trace(`DriverHelper.waitVisibility ${elementLocator}`);
 
         for (let i = 0; i < attempts; i++) {
-            Logger.trace('DriverHelper.waitVisibility - Waiting until element is located');
-            const webElement: WebElement = await this.driver.wait(until.elementLocated(elementLocator), timeout);
-            Logger.trace('DriverHelper.waitVisibility - Element was located. Waitng for visibility now.');
             try {
```

Comment:
I have a concern about two levels of the try/catch, in my opinion, this one is redundant.

Reply:
I had issues for the methods failing after times that did not match the timeout without a sensual exception message.
The waitVisibility method was also failing with ElementNotFound that was never caught and caused issue with the logic of the tests.
The second layer of try-catch is there to prevent the method failing without reaching the desired amount of attempts/timeout and if the issue persists, it throws the correct exception.

Reply:
But it was working stable with the previous version, am I right? 
I mean if yes then you should focus on the changes and find out a reason and ideally fix a reason instead of wrapping it by "try/catch".
Even if you will decide to use "try/catch", in my eyes, if you have an issue with one line, just wrap this single line by "try/catch" instead of using nested "try/catch".
In my opinion, this way will simplify the readability of the code.

Reply:
I've split it into two separate try-catch blocks